funnel/models/event.py

```diff
@@ -136,32 +155,18 @@ def attendees_by_event(cls, event):
         return db.session.execute(stmt).fetchall()
 
     @classmethod
-    def make_from_dict(cls, participant_dict, space):
-        return Participant(
-            fullname=participant_dict.get('fullname'),
-            email=participant_dict.get('email'),
-            phone=participant_dict.get('phone'),
-            twitter=participant_dict.get('twitter'),
-            job_title=participant_dict.get('job_title'),
-            company=participant_dict.get('company'),
-            city=participant_dict.get('city'),
-            proposal_space=space
-        )
-
-    @classmethod
-    def update_from_dict(cls, updated_participant):
-        participant = Participant.query.filter_by(email=updated_participant.get('email')).first()
+    def update_from_dict(cls, space, fields):
+        participant = cls.get(space, fields.get('email'))
         if participant:
-            participant.fullname = updated_participant.get('fullname')
-            participant.phone = updated_participant.get('phone')
-            participant.twitter = updated_participant.get('twitter')
-            participant.job_title = updated_participant.get('job_title')
-            participant.company = updated_participant.get('company')
-            participant.city = updated_participant.get('city')
-            db.session.add(participant)
+            participant.fullname = fields.get('fullname')
+            participant.phone = fields.get('phone')
+            participant.twitter = fields.get('twitter')
+            participant.job_title = fields.get('job_title')
+            participant.company = fields.get('company')
+            participant.city = fields.get('city')
             return participant
         else:
-            return None
+            raise NoResultFound
```

Comment:
1. `NoResultFound` implies you were searching. `update` in the method name doesn't sound like searching.
2. No good reason for this to be a class method. Let `get(create=True)` return an empty participant and `update_from_dict` be an instance method that applies on the current instance.